### PR TITLE
Major functionality updates and bug fixes

### PR DESCRIPTION
**Pull request by bunsenmurder**
What's been done:
- Upgraded threading to concurrent.futures
- Implemented delaying algorithm to calculate delays. I had to separate each child classes blurb retrieval classes into two which were passed as arguments to the `delay_threader` method in the parent class.
- Optimized all regex and added all possible cases for date string capture, also the date is saved as YYYY-MM-DD now.
- tfidf filter now filters on `scrape.data`  itself on first scrape, and before comparing to the master_list on subsequent  scrapes.
- Added the `id_filter()` method to _filters.py_ that filters for ids found in the master_list and duplicate_list.
- Updated glassdoor radius compatibility, and added many similar functions to other provider classes with compatibility for .ca and .com domains.
- Added the `pre_filter` method to the `JobFunnel` class, that is called by the child classes and filters anything that it can before requesting blurbs. 
- Bug fixes

I have done a lot of other stuff too, but if I write all of them out this will turn into a book. My commit messages are pretty detailed about what was done, and I wrote lots of comments in the code. If you need more detailed explanations, please let me know.

### CODE REVIEW

**Comment by bunsenmurder** [comment]
I also have two questions:
1. How would we handle versioning? I'm pretty sure everything that I have done would constitute a whole version bumb.
2. How would you want handle updating the readme?

**Comment by PaulMcInnis** [comment]
Looks great, RE the two questions.

1. Feel free to bump the major version in the init file. 
2. For the readme, in usage notes, lets mention configuring the respectful scraping delay and any other usage changes.

**Comment by bunsenmurder** [comment]
Will do and do you mind if I keep the link to the [Jupyter Notebook](https://github.com/bunsenmurder/Notebooks/blob/master/jobFunnel/delay_algorithm.ipynb) on the delaying algorithm in the readme? I will edit it, of course.

**Comment by PaulMcInnis** [comment]
Yeah for sure link to it, fits in nicely with our other demo.

**Comment by bunsenmurder** [comment]
Alright, I updated everything. Let me know, if you need anything!

File: jobfunnel/tools/filters.py [hunk 1]
```diff
@@ -1,59 +1,124 @@
...
             list of duplicate job ids which were removed from cur_dict
-
-        @TODO skip calculating metric for jobs which have the same job id!
     """
+    # Retrieve stopwords if not already downloaded.
+    try:
+        stopwords = nltk.corpus.stopwords.words('english')
+    except LookupError:
+        try:
+            nltk.download('stopwords', quiet=True)
```
Comment (PaulMcInnis) [inline jobfunnel/tools/filters.py]: Need to assign stopwords after downloading here as well
`stopwords = nltk.corpus.stopwords.words('english')`

Comment (bunsenmurder) [reply jobfunnel/tools/filters.py]: Good catch, I just put a fix out for it.

**Pull request merged**